src/vcpkg/archives.cpp

```diff
@@ -191,6 +201,28 @@ namespace
 
 namespace vcpkg
 {
+    void extract_self_extracting_7z(const VcpkgPaths& paths, const Path& archive, const Path& to_path)
+    {
+        constexpr static const char header_7z[] = "7z\xBC\xAF\x27\x1C";
+
+        const Path stem = archive.stem();
+        const auto subext = stem.extension();
+        Checks::check_exit(VCPKG_LINE_INFO,
+                           Strings::case_insensitive_ascii_equals(subext, ".7z"),
+                           "Unable to extract 7z archive from Installer %s. Missing '.7z.exe' extension.",
+                           archive);
+
```

Comment:
Similar to my comment in the other PR I don't think we should treat .7z.exe differently than .exe.

Reply:
why not? vcpkg treats a number of extension differently here so treating sub-extensions differently is just the same. Since vcpkg is in control of the file name I don't really see an issue with that.

Reply:
also the other PR is just for exactly one tool (7z). This PR is for multiple tools having a `.7z.exe` installer package. (although it currently only applies to PortableGit since it is the only 7z installer package).
So the scope and application is a bit different to the other PR.

Reply:
I'm saying we should be interpreting all .exe as .7z.exe, because that is true for all the .exes we support passing to this right now. In the other example you checked the whole file name rather than just the extension in order to determine that we had a 7zip installer.

Reply:
Ok moved everything into the `.exe` path but I want to keep the requirement that the file is named `.7z.exe` before the extraction